src/app/clusters/level-control/level-control.cpp

```diff
@@ -139,6 +141,27 @@ static void timerCallback(System::Layer *, void * callbackContext)
     emberAfLevelControlClusterServerTickCallback(static_cast<EndpointId>(reinterpret_cast<uintptr_t>(callbackContext)));
 }
 
+static uint32_t calculateNextTimestampMs(uint32_t delayMs)
+{
+    int32_t waitTime, latency, delay = (int32_t) delayMs;
+    const chip::System::Clock::Timestamp currentTime = chip::System::SystemClock().GetMonotonicTimestamp();
+
+    latency = (int32_t) currentTime.count() - (int32_t) nextTimestamp.count();
+
+    if (latency > delay)
+    {
+        waitTime = 0;
```

Comment:
This feels like we can end up flooding the event loop completely if handling a single callback from this timer takes longer than `delayMs` in general....  Are we fairly certain that will not happen?